### PR TITLE
Implements createIndexes command.

### PR DESCRIPTION
**Pull request by vladlosev**
@parkr @rodaine This ~~implements~~fakes index creation using the `createIndexes` command and the direct insertion into `system.indexes` (this is how the mongo command line client does it, BTW - feel free to play with it from the client).

### CODE REVIEW

File: lib/commands/base.js [hunk 1]
```diff
@@ -22,7 +22,14 @@ BaseCommand.prototype.getDbAndCollectionName = function(clientReqMsg) {
...
    if (collectionName === '$cmd') {
-     collectionName = clientReqMsg.query[this.commandName];
+     fullCollectionName = clientReqMsg.query[this.commandName];
+     firstDot = fullCollectionName.indexOf('.');
+     if (firstDot === -1) {
+       collectionName = fullCollectionName;
+     } else {
+       databaseName = fullCollectionName.substring(0, firstDot);
+       collectionName = fullCollectionName.substring(firstDot + 1);
+     }
```
Comment (parkr) [inline lib/commands/base.js]: isn't this behaviour like `getCollection()`? would you want to abstract this out?

File: lib/processor.js [hunk 2]
```diff
@@ -85,6 +86,12 @@ Processor.prototype.getCollection = function getCollection(
   return Processor._mocks[databaseName][collectionName] || [];
 };
 
+Processor.prototype.collectionExists = function collectionExists(
+  databaseName,
+  collectionName) {
```
Comment (parkr) [inline lib/processor.js]: this is kind of awkward spacing, i thought it was in the body of the function.

**Comment by parkr** [comment]
:tada: :shipit:

File: lib/commands/base.js (hunk 1 above)
Comment (vladlosev) [inline lib/commands/base.js]: Well, this *is* the helper that `getCollection` uses. :smiley_cat:

**Comment by rodaine** [comment]
Looks good. Is this a step towards [`createIndex`](http://docs.mongodb.org/manual/reference/method/db.collection.createIndex/#db.collection.createIndex) (and the [now defunct `ensureIndex`](http://docs.mongodb.org/manual/reference/method/db.collection.ensureIndex/))?

**Comment by vladlosev** [comment]
The MongoDB node driver [implements](https://github.com/mongodb/node-mongodb-native/blob/81ccf206ab1b10d7aefc570bebef4ca8b86b0071/lib/db.js#L680) both `createIndex` and `ensureIndex` using this command and direct insertion into `system.indexes` as a backup. Both ways are implemented in this PR.

File: lib/processor.js (hunk 2 above)
Comment (vladlosev) [inline lib/processor.js]: Agree. I have increased indent here for the parameter list in order to visually separate the function parameters and the body.

**Pull request merged**